### ISSUE TITLE
TypeScript: RoutableProps lacks matches, and shouldn't be added to all components

### ISSUE THREAD

**Issue by prencher** [developit/preact-router#269]
* The RoutableProps interface is lacking the `matches` key.
* RoutableProps is being added to _everything_. This is deeply incorrect, and should be corrected as soon as possible, as it makes the types wrong. #244 is naive in suggesting this should be the case.

**Comment by Alexendoo** [comment]
What would this matches key be exactly?

As for adding to PreactHTMLAttributes - it works well in practise, is there a specific issue it's causing?

**Comment by prencher** [comment]
The matches key is a `Record<string, string>`, I assume. It's in the documentation, and always present.

The issue that adding it to all components causes is that it's flat out incorrect. These are not props my components take, and they are only provided to the immediate children of `Route` and `Match`.

I can see an argument for keeping it on plain HTML elements, since you can't really control the interfaces on those, if possible.

I use TypeScript because I want type correctness. It's problematic that a library is arbitrarily messing with this expectation.

**Comment by Alexendoo** [comment]
Ah I see, I'm not sure how to type matches exactly, it's not something that should be passed as a property yourself so it would have to be optional but that's still a bit misleading

I'm still in favour of adding it to PreactHTMLAttributes personally, it's a worthwhile trade-off in my mind to be able to use the library (mostly) as documented in typescript without requiring each component to depend on preact-router

**Comment by prencher** [comment]
I disagree with that, but as long as the missing `matches` is covered, it'd at least represent the correct API.

**Comment by developit** [comment]
I recall TS recently added support for specifying props for a child component, but I've lost the link.  That seems like it would be relevant here, where we want to mix RoutableProps into the Prop types of any child of a Router.
